nav/coordinates.py

```diff
@@ -75,28 +84,89 @@ def finalize(self):
             self.crossings  = np.where(np.diff(detect))[0]
             self.day        = ~detect
 
-            # Place for future logic dev
-            #if self.day[0]:
-            #    for cr in self.crossings
+
+        self.mjd_crossings = np.fromiter(self.precise_crossings(), float)
         self.set_temperature()
 
-    ###
-    def calculate(self, interval):
-        # Note the crafty logic in the Observation class constructor - it's hand-made polymorphism.
-        # After update in Nov 2023, the Observation ctor can take a tuple of start and end time points.
+
+    ### ---
+    def clock(self, mjd):
+        """ This method calculates the "lunar clock" e.g. the time according
+            to 24-hour subdividion of the Lunar day.
+        """
         
+        if mjd>self.mjd[self.N-1] or mjd<self.mjd[0]: return None
+
+        if mjd<self.mjd_crossings[0]: # only one endpoint at the start, use the next cycle as estimate
```

Comment:
I don't have time to go through this hocus-pocus below, but let's assume it is right for now and we can fix it later. :)

Reply:
It works. For the outmost points on left and right over the prefab data series, we have to arrive to an estimate of the duration of day or night, because there is only one boundary. There is nothing else that we can do except the simple step which is to start and stop the sim at the points where there is always transits on left and right. In the current application, this will hardly matter.